qiskit/visualization/__init__.py

```diff
@@ -17,20 +17,160 @@
 
 .. currentmodule:: qiskit.visualization
 
+
+Overview
+========
+
+The visualization module contain functions that visualizes measurement outcome counts, quantum
+states, circuits, pulses, devices and more.
+
+Prerequisites
+=============
+
+To use visualization functions, you are required to install visualization optionals to your
+development environment:
+
+.. code-block:: bash
+
+   pip install 'qiskit[visualization]'
+
 Counts and State Visualizations
 ===============================
 
+This section contains functions that visualize measurement outcome counts and quantum states.
+
+APIs
+----
+
+Counts Visualization
+^^^^^^^^^^^^^^^^^^^^
+
 .. autosummary::
    :toctree: ../stubs/
 
    plot_histogram
+
+State Visualization
+^^^^^^^^^^^^^^^^^^^
+
+.. autosummary::
+   :toctree: ../stubs/
+
    plot_bloch_vector
    plot_bloch_multivector
    plot_state_city
    plot_state_hinton
    plot_state_paulivec
    plot_state_qsphere
 
+Example Usage
+-------------
+
+Counts Visualization
+^^^^^^^^^^^^^^^^^^^^
+
+Here is an example of using :func:`plot_histogram` to visualize measurement outcome counts:
+
+.. jupyter-execute::
+
+   from qiskit.tools.visualization import plot_histogram
```

Comment:
Didn't notice that on my first read, but should we import `qiskit.tools.visualization` or `qiskit.visualization`?

Reply:
Good point. Look at the PR migrating the file, I think we should use `qiskit.visualization`. https://github.com/Qiskit/qiskit-terra/pull/1878